proto/clustermanagerdata.proto

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 The Vitess Authors
+ Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreedto in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file contains vitess cluster manager related data structures.
+
+syntax = "proto3";
+option go_package = "vitess.io/vitess/go/vt/proto/clustermanagerdata";
+
+option java_package="io.vitess.proto";
+
+import "topodata.proto";
+
+package clustermanagerdata;
+
+// SetShardConfigRequest is the payload for SetShardConfigRequest
+message SetShardConfigRequest {
+  string cell = 1;
+  string keyspace = 2;
+  string shard = 3;
+  topodata.TabletType tablet_type = 4;
+  // number of nodes desired for (keyspace/shard/cell/tablet_type)
+  int32 nodes = 5;
```

Comment:
So if I want to setup a new shard with 1 master and 4 replicas and 8 rdonlys, I have to issue 3 SetShardConfigRequests?

Reply:
Yeah, right now that is correct. I was optimizing for the normal case when you are just updating an existent shard but not when you are creating new ones. 

But I think creating new shards will be very annoying this way. How about we create a new RPC call that accepts `ClusterConfig` as a parameter and it only works for new cells/keyspaces/shards? It would fail for existent ones, as I would be too risky to update in bulk those.

Reply:
I think this is the right API to have for this. Its the primitive and we can build on top of it for making complex changes.